operator/v1/types_dns.go

```diff
@@ -170,6 +235,15 @@ type ForwardPlugin struct {
 	// +optional
 	// +kubebuilder:default:="Random"
 	Policy ForwardingPolicy `json:"policy,omitempty"`
+
+	// transportConfig is used to configure the transport type, server name, and optional custom CA or CA bundle to use
+	// when forwarding DNS requests to an upstream resolver.
+	//
+	// The default value is "" (empty) which results in a standard cleartext connection being used when forwarding DNS
+	// requests to an upstream resolver.
+	//
+	// +optional
```

Comment:
Any reason not to use something like `// +kubebuilder:default={"transport":"Cleartext"}`?

Reply:
Initially, I had this set like you've suggested but @deads2k raised a good point. If the default is `Cleartext` it's ambiguous if the cluster is running the default `"" (empty)` or if the cluster admin chose `Cleartext` specifically (ex. going from DNS-over-TLS to Cleartext). I thought there was a comment in this PR with this discussion but I can't find it; may have been another communication mechanism. I'll count on @deads2k and @JoelSpeed to keep me honest.

Reply:
In that case, I think you'll need to include the empty string as an allowed value for the `DNSTransport` type:
```go
// +kubebuilder:validation:Enum=TLS;Cleartext;""
type DNSTransport string
```

Reply:
I think the intention of leaving this empty by default is so that, in the future, we can change the default. 

If we use openapi defaulting, then in the future we won't be able to change the opinionated default and transparently migrate users from cleartext to tls, we would need the end users to explicitly choose to migrate which I believe was previously dismissed, in favour of us eventually migrating users _unless_ they explicitly choose Cleartext

> In that case, I think you'll need to include the empty string as an allowed value for the DNSTransport type:

I just tested this (I wasn't sure), but if you omit the field (because it's optional), that's fine, if you explicitly set it to `""` then you need to have it in the enum, so probably worth adding